src/scripts.c

```diff
@@ -34,45 +34,59 @@
 
 static void update_kernel(void)
 {
-	char *kernel_update_cmd = NULL;
-
 	if (strcmp("/", path_prefix) == 0) {
-		string_or_die(&kernel_update_cmd, "/usr/bin/kernel_updater.sh");
+		char *const kernel_update_cmd[] = { "/usr/bin/kernel_updater.sh", NULL } ;
+
+		system_argv(kernel_update_cmd);
 	} else {
-		string_or_die(&kernel_update_cmd, "%s/usr/bin/kernel_updater.sh --path %s", path_prefix, path_prefix);
-	}
+		char *pathupdater;
+
+		string_or_die(&pathupdater, "%s/usr/bin/kernel_updater.sh", path_prefix);
+		char *const kernel_update_cmd[] = { pathupdater, "--path", path_prefix,  NULL } ;
 
-	system(kernel_update_cmd);
-	free(kernel_update_cmd);
+		system_argv(kernel_update_cmd);
+		free(pathupdater);
+	}
 }
 
 static void update_bootloader(void)
 {
-	char *bootloader_update_cmd = NULL;
-
 	if (strcmp("/", path_prefix) == 0) {
-		string_or_die(&bootloader_update_cmd, "/usr/bin/gummiboot_updaters.sh");
+		char *const bootloader_update_cmd[] = { "/usr/bin/gummiboot_updaters.sh", NULL };
```

Comment:
Unrelated to your patch, but the gummiboot_updaters.sh call can be removed anytime, since it's not used anymore.